api/resource.rb

```diff
@@ -151,6 +153,18 @@ def validate
       end
     end
 
+    # Represents a list of documentation links.
+    class IamPolicy < Api::Object
+      # boolean of if this binding should be generated
+      attr_reader :exists
```

Comment:
Do we even need an `exists` field? Presence of the object should be sufficient for now, unless we find we need to override it later.  (And if we do, I'd be in slight preference of `exclude` for consistency with top-level resources)

Reply:
True. Just feels weird to have it be an empty object at this point

Reply:
Fun YAML fact: you can't have an empty object. Every Ruby object, when serialized to YAML, needs at least one field. Yes, it's dumb.

The way we've handled this is that any thing with < 1 field is just a boolean instead of a full Ruby value.

Reply:
🤦‍♀️ what a silly rule. Are we able to add a single field and then not use it anywhere?

Reply:
They might've actually changed that? I remember that being an issue early on.

test code:
```ruby
require 'yaml'

class Test
end

puts YAML.dump(Test.new)
```

output:
```yaml
--- !ruby/object:Test {}
```

Looks like the {} syntax would make it possible to have a blank object. Not the prettiest thing in the world, but what are you going to do?

Reply:
You could have an `exclude` here. So that a provider could have the top level resource but not the IAM associated.

Reply:
Yep- if we need a single field imo it should be `exclude`